DPL Analysis: write HistogramRegistry incrementally

ktf opened the pull request:
This should reduce big spikes at the end of the processing when a large
HistogramRegistry is serialised.

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/AliceO2Group/AliceO2/pull/14646).
* __->__ #14646
* #14692

alibuild commented:
Error while checking build/O2/fullCI_slc9 for daa8075475e5b624dc6dcaddeae7e891a419dcbe at 2025-09-18 15:18:
```
## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile reco_NOGPU.log


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
grep: error-log.txt: binary file matches
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/7e6f7431dfd5802267616562ce8df00fe0b99938/slc9_x86-64/o2checkcode/1.0-local328/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14646/daa8075475e5b624dc6dcaddeae7e891a419dcbe/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for d20604b796a0f7f8ebd70339daef3ef64ba45720 at 2025-09-22 19:11:
```
++ cp /sw/slc9_x86-64/O2/14646-slc9_x86-64-local1/compile_commands.json .
+++ python3 -c 'import json, os; print(os.path.commonpath([x["file"] for x in json.loads(open("compile_commands.json").read()) if "sw/BUILD" not in x["file"] and "G__" not in x["file"] and x["file"].endswith(".cxx")]))'
++ O2_SRC=/sw/SOURCES/O2/14646-slc9_x86-64/0
++ [[ -e /sw/SOURCES/O2/14646-slc9_x86-64/0/CMakeLists.txt ]]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14646/d20604b796a0f7f8ebd70339daef3ef64ba45720/build_O2_fullCI_slc9/pretty.html).

ktf committed:
DPL: allow closing a signpost interval with an error

The error will be printed regardless of the signposts being enabled.

In case the signposts are actually enabled, the error will be the closing
message of the signpost interval.

In case the signposts are not enabled, the error will be printed as a standard error.

ktf committed:
DPL Analysis: write HistogramRegistry incrementally

This should reduce big spikes at the end of the processing when a large
HistogramRegistry is serialised.
